ecr-login/ecr.go

```diff
@@ -70,14 +71,36 @@ func NewECRHelper(opts ...Option) *ECRHelper {
 // ensure ECRHelper adheres to the credentials.Helper interface
 var _ credentials.Helper = (*ECRHelper)(nil)
 
-func (ECRHelper) Add(creds *credentials.Credentials) error {
-	// This does not seem to get called
-	return notImplemented
+func shouldIgnoreCredsStorage() bool {
+	return os.Getenv("AWS_ECR_IGNORE_CREDS_STORAGE") == "true"
 }
 
-func (ECRHelper) Delete(serverURL string) error {
-	// This does not seem to get called
-	return notImplemented
+// Add tries to store credentials when docker requests it. This usually happens during `docker login` calls. In our context,
+// storing arbitrary user given credentials makes no sense.
+func (self ECRHelper) Add(creds *credentials.Credentials) error {
+	if shouldIgnoreCredsStorage() {
+		self.logger.
+			WithField("serverURL", creds.ServerURL).
+			Warning("Ignoring request to store credentials since AWS_ECR_IGNORE_CREDS_STORAGE env variable is set." +
+				"This is not supported in the context of the docker ecr-login helper.")
+		return nil
+	} else {
+		return notImplemented
+	}
```

Comment:
Consider adding a message here too that alerts the user that `AWS_ECR_IGNORE_CREDS_STORAGE` should be set to allow the process to succeed in this case.

```suggestion
	logger := self.logger.WithField("serverURL", creds.ServerURL)
	
	if shouldIgnoreCredsStorage() {
		logger.Warning("Ignoring request to store credentials since AWS_ECR_IGNORE_CREDS_STORAGE env variable is set." +
				"ecr-login does not require 'docker login', and does not support persisting temporary ECR-issued credentials.")
		return nil
	} else {
		logger.Warning("Saving credentials is not supported by the ecr-login credentials helper as all issued credentials are temporary. Consider setting the AWS_ECR_IGNORE_CREDS_STORAGE env variable (see documentation for details).")
		return notImplemented
	}
```

Reply:
This lines up nicely with what I was thinking about [here](https://github.com/awslabs/amazon-ecr-credential-helper/pull/847#discussion_r1697718013) and does a better job at explaining things. I will add this in, thanks!